### PR TITLE
Add Feature Extraction Code

### PR DESCRIPTION
**Pull request by yashkant**

### CODE REVIEW

File: captioning/utils/image_preproc.py [hunk 1]
```diff
@@ -1,25 +1,27 @@
...
 import requests
+import math
 from PIL import Image
 from maskrcnn_benchmark.layers import nms
 from maskrcnn_benchmark.structures.image_list import to_image_list
 
 # TODO: Comments, Docstrings, Cleanup.
 
-def get_actual_image(image_path):
+def read_actual_image(image_path):
```
Comment (abhshkdz) [inline captioning/utils/image_preproc.py:13]: `read_image` instead of `read_actual_image`?

File: captioning/utils/image_preproc.py [hunk 2]
```diff
@@ -1,25 +1,27 @@
...
+    return img
 
 
 def image_transform(image_path):
-    path = get_actual_image(image_path)
-    img = Image.open(path)
+    img = read_actual_image(image_path)
     im = np.array(img).astype(np.float32)
     im = im[:, :, ::-1]
     im -= np.array([102.9801, 115.9465, 122.7717])
```
Comment (abhshkdz) [inline captioning/utils/image_preproc.py:27]: Add a comment here please on which model these constants are for? Will make life easier in future.

File: captioning/utils/image_preproc.py [hunk 3]
```diff
@@ -95,7 +99,7 @@ def process_feature_extraction(output,
             )
 
         # TODO: add max-boxes config instead of 100 and use below
```
Comment (abhshkdz) [inline captioning/utils/image_preproc.py:101]: Drop? Incorporated already, right?

File: captioning/utils/image_preproc.py [hunk 4]
```diff
@@ -105,29 +109,24 @@ def process_feature_extraction(output,
...
-        # TODO: Use better arg-name and loop vars
-        for img_path in image_path:
-            im, im_scale = image_transform(img_path)
-            img_tensor.append(im)
-            im_scales.append(im_scale)
-    else:
-        im, im_scale = image_transform(image_path)
-        img_tensor, im_scales = [im], [im_scale]
+    for img_path in image_paths:
+        im, im_scale = image_transform(img_path)
```
Comment (abhshkdz) [inline captioning/utils/image_preproc.py:124]: Can we change this to
```
im = read_image(img_path)
im, im_scale = transform_image(im)
```
Feels odd for `transform` to also be reading the image inside it.

File: captioning/utils/image_preproc.py [hunk 5]
```diff
@@ -138,12 +137,28 @@ def get_detectron_features(image_path,
         feat_name=feat_name,
         conf_thresh=0.2
     )
+    return return_list
```
Comment (abhshkdz) [inline captioning/utils/image_preproc.py:140]: Should we pick a more informative variable name than `return_list`?

File: captioning/utils/image_preproc.py (hunk 3 above)
Comment (yashkant) [reply captioning/utils/image_preproc.py:101]: Yes, dropped.

File: captioning/utils/image_preproc.py (hunk 4 above)
Comment (yashkant) [reply captioning/utils/image_preproc.py:124]: Done!

File: captioning/utils/image_preproc.py (hunk 5 above)
Comment (yashkant) [reply captioning/utils/image_preproc.py:140]: Changed `feat_list`

**Pull request merged**